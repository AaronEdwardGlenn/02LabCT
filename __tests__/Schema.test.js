const Schema = require('../lib/Schema.js');

describe('Schema', () => {
    it('validate a good schema', () => {
        const schema = new Schema({
            name: {
                type: String,
                required: true
            }, 
            age: {
                type: Number, 
            }, 
            weight: {
                type: String, 
            }
        });

        const dog = {
            name: 'scruffy',
            age: 69, 
            weight: '420 lbs'
        }; 

        expect(schema.validate(dog)).toEqual({
            name: 'scruffy', 
            age: 69,
            weight: '420 lbs'
        }); 
    });

    it('throws on a bad schema', () => {
        const schema = new Schema({
            name: {
                type: String, 
                required: true
            }, 
            age: {
                type: Number
            }, 
            weight: {
                type: String
            }
        });
        const dog = {
            age: 'scruffy',
            weight: '420 lbs'
        };

        expect(() => schema.validate(dog)).toThrowErrorMatchingSnapshot(); 
    });
});