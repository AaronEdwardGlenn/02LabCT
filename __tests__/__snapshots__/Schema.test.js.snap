// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema throws on a bad schema 1`] = `"this schema is not even close to valid. Sort your life out >> Error: name needs to be here. What are you doing. Serously.,Error: Cannot cast >>scruffy<< to Number"`;
